### PR TITLE
emrun establish browser url after starting web server

### PR DESCRIPTION
**Pull request by arsnyder16**
This allows port 0 server binding to work with browser launching.

### CODE REVIEW

File: emrun.py [hunk 1]
```diff
@@ -1643,10 +1643,6 @@ def run():
     url = file_to_serve
   else:
     url = os.path.relpath(os.path.abspath(file_to_serve), serve_dir)
```
Comment (sbc100) [inline emrun.py:1645]: Maybe move these 4 lines down too do avoid have two separate `if file_to_serve_is_url` block.

File: emrun.py [hunk 2]
```diff
@@ -1659,6 +1655,13 @@ def run():
     # to support binding to port zero we must allow the server to open to socket then retrieve the final port number
     options.port = httpd.socket.getsockname()[1]
 
+  if not file_to_serve_is_url:
+    if len(options.cmdlineparams):
+      url += '?' + '&'.join(options.cmdlineparams)
+    hostname = socket.gethostbyname(socket.gethostname()) if options.android else options.hostname
+    # to support binding to port zero we must allow the server to open to socket then retrieve the final port number
```
Comment (sbc100) [inline emrun.py:1662]: I don't think there is a need to repeat this comment, or at least it should be re-phrased to better apply to this block.

**Review by sbc100 — COMMENTED** [review]
What is the net effect of this change?  i.e. what is `url` later used for here?    Can we write a test for it?

File: emrun.py (hunk 1 above)
Comment (arsnyder16) [reply emrun.py:1645]: Yea so I tried this but the combination of relative and absolute paths to the file to server didn’t play well

**Comment by arsnyder16** [comment]
> What is the net effect of this change? i.e. what is `url` later used for here? Can we write a test for it?

The url is used for launching the browser so if you say bind to port 0, the web server will bring to an available port and we cannot tell browser to navigate to localhost:0

Do we have tests around emrun port argument? If so I am happy to add 0 to those test cases

File: emrun.py (hunk 2 above)
Comment (arsnyder16) [reply emrun.py:1662]: I am open to suggestions, this seems appropriate to me

**Review by sbc100 — APPROVED** [review]

**Comment by arsnyder16** [comment]
@sbc100 Would it be possible to have this land in in `3.1.35`?

**Pull request merged**